src/v/pandaproxy/kafka_client_cache.cc

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "pandaproxy/kafka_client_cache.h"
+
+#include "pandaproxy/logger.h"
+
+namespace pandaproxy {
+
+kafka_client_cache::kafka_client_cache(
+  YAML::Node const& cfg, size_t max_size, std::chrono::milliseconds keep_alive)
+  : _config{cfg}
+  , _cache_max_size{max_size}
+  , _keep_alive{std::move(keep_alive)} {}
+
+client_ptr kafka_client_cache::make_client(
+  credential_t user, config::rest_authn_method authn_method) {
+    kafka::client::configuration cfg{
+      to_yaml(_config, config::redact_secrets::no)};
+
+    // Set the principal when the request is using HTTP Basic AuthN
+    if (authn_method == config::rest_authn_method::http_basic) {
+        // Need to specify type or else bad any_cast runtime error
+        cfg.sasl_mechanism.set_value(ss::sstring{"SCRAM-SHA-256"});
+        cfg.scram_username.set_value(user.name);
+        cfg.scram_password.set_value(user.pass);
+    }
+
+    return ss::make_lw_shared<kafka::client::client>(
+      to_yaml(cfg, config::redact_secrets::no));
```

Comment:
Only use `redact_secrets::yes` prior to logging because it will cause authn/authz problems on the kafka api. Otherwise use `redact_secrets::no`.

More details here https://github.com/redpanda-data/redpanda/pull/6452#discussion_r974469365